src/server/worker/cache/cache.go

```diff
@@ -0,0 +1,60 @@
+package cache
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/pachyderm/pachyderm/src/server/pkg/hashtree"
+)
+
+// WorkerCache is an interface for managing hashtree MergeCaches for multiple
+// concurrent jobs
```

Comment:
Looking at the implementation it seems like this is unsafe for concurrent access. We should probably note that in the comment above, or if it makes sense we should make it safe for concurrent access.